sdk/src/main/scala/com/horizen/account/storage/AccountStateMetadataStorageView.scala

```diff
@@ -0,0 +1,350 @@
+package com.horizen.account.storage
+
+import com.google.common.primitives.{Bytes, Ints}
+import com.horizen.account.receipt.{EthereumReceipt, EthereumReceiptSerializer}
+import com.horizen.account.storage.AccountStateMetadataStorageView.DEFAULT_ACCOUNT_STATE_ROOT
+import com.horizen.account.utils.{AccountBlockFeeInfo, AccountBlockFeeInfoSerializer}
+import com.horizen.block.{WithdrawalEpochCertificate, WithdrawalEpochCertificateSerializer}
+import com.horizen.consensus.{ConsensusEpochNumber, intToConsensusEpochNumber}
+import com.horizen.storage.Storage
+import com.horizen.utils.{ByteArrayWrapper, WithdrawalEpochInfo, WithdrawalEpochInfoSerializer, Pair => JPair, _}
+import sparkz.core._
+import scorex.crypto.hash.Blake2b256
+import scorex.util.ScorexLogging
+import java.math.BigInteger
+import java.util.{ArrayList => JArrayList}
+import scala.collection.mutable.ListBuffer
+import scala.compat.java8.OptionConverters._
+import scala.util.{Failure, Success, Try}
+
+
+class AccountStateMetadataStorageView(storage: Storage) extends AccountStateMetadataStorageReader with ScorexLogging {
+
+  require(storage != null, "Storage must be NOT NULL.")
+
+  private[horizen] val ceasingStateKey = calculateKey("ceasingStateKey".getBytes)
+  private[horizen] val heightKey = calculateKey("heightKey".getBytes)
+  private[horizen] val withdrawalEpochInformationKey = calculateKey("withdrawalEpochInformation".getBytes)
+  private[horizen] val consensusEpochKey = calculateKey("consensusEpoch".getBytes)
+  private[horizen] val accountStateRootKey = calculateKey("accountStateRoot".getBytes)
+  private[horizen] val baseFeeKey = calculateKey("baseFee".getBytes)
+
+  private val undefinedBlockFeeInfoCounter: Int = -1
```

Comment:
Should be capitalized and moved to AccountStateMetadataStorageView companion obj